### ISSUE TITLE
Sam's Thesis Meeting | Jan 29, 2023 - 10:30am - ITB 167

### ISSUE THREAD

**Issue by samm82** [samm82/TestGen-Thesis#31]
My work this past week has mainly been going through the IEEE sources a second time with a finer-toothed comb to get more information, closing #27 and working towards #25. While these items are slightly less "meatier" than what we've discussed before, here are some things I would want to talk about (potentially as well as anything of note on your end):

1. The possibility of my Master's going into the summer. What would that look like? I'm assuming that a literature review doesn't constitute a Master's project; with this assumption, I doubt I would be finished by April.
2. From my [degree requirements](https://www.cas.mcmaster.ca/cas/0files/reg_master_se_2019.pdf), what would a technical presentation look like? I'm assuming it wouldn't be enough to do this just based on my testing literature review, but I also don't know if I'd have much more done by the competition date (see point 1.)
3. Potential next steps (extra info: [SWEBOK v4 is under review and looking for comments](https://www.computer.org/volunteering/boards-and-committees/professional-educational-activities/software-engineering-committee/swebok-evolution?vgo_ee=PyjiD1O%2FIknuvXKdnx5H%2BgU34azCWzF5AhE7wuTCxyGmd3I%3D:LJvnpqfYGmFPgco6rLvVbh45sfCemRyW)):
   a. Finish going through IEEE2017
   b. Finish the "second pass" of what I've already done of IEEE2017, then pivot to SWEBOK
   c. Go through SWEBOK (v3, the current one)
   d. Go through SWEBOK (v4, under review) - would this be a reasonable source to cite, or would this have to be slightly separated from the rest of my research
   e. Go through SWEBOK v3 _while_ going through v4 to compare/contrast 
4. Acceptance Tests vs. Acceptance Testing (see [IEEE2017](https://gitlab.cas.mcmaster.ca/crawfs1/TestGen-Thesis-Papers/-/blob/main/IEEE2017.pdf?ref_type=heads)) - is this a level of distinction we should make? Is this relevant?

**Comment by smiths** [comment]
We can discuss this in more detail during the meeting, but I'll put down some initial thoughts:

> My work this past week has mainly been going through the IEEE sources a second time with a finer-toothed comb to get more information, closing #27 and working towards #25. While these items are slightly less "meatier" than what we've discussed before, here are some things I would want to talk about (potentially as well as anything of note on your end):
> 
> 1. The possibility of my Master's going into the summer. What would that look like? I'm assuming that a literature review doesn't constitute a Master's project; with this assumption, I doubt I would be finished by April.

It would be great to finish your Masters by April or May, but that doesn't seem likely.  Most students take 24 months, so working over the summer seems reasonable.  We can discuss further in our meeting.

> 2. Clarifications about my [degree requirements](https://www.cas.mcmaster.ca/cas/0files/reg_master_se_2019.pdf)
>    a. Does the departmental seminar attendance apply to every year, or just the first year?

I read the statement as seminar attendance is only necessary during the first year, although it would be nice to attend more often than required.  :smile:

>    b. What would a technical presentation look like? I'm assuming it wouldn't be enough to do this just based on my testing literature review, but I also don't know if I'd have much more done by the competition date (see point 1.)

Yes, a well done literature review can be the topic of a technical poster.

> 3. Potential next steps (extra info: [SWEBOK v4 is under review and looking for comments](https://www.computer.org/volunteering/boards-and-committees/professional-educational-activities/software-engineering-committee/swebok-evolution?vgo_ee=PyjiD1O%2FIknuvXKdnx5H%2BgU34azCWzF5AhE7wuTCxyGmd3I%3D:LJvnpqfYGmFPgco6rLvVbh45sfCemRyW)):
>    a. Finish going through IEEE2017
>    b. Finish the "second pass" of what I've already done of IEEE2017, then pivot to SWEBOK
>    c. Go through SWEBOK (v3, the current one)
>    d. Go through SWEBOK (v4, under review) - would this be a reasonable source to cite, or would this have to be slightly separated from the rest of my research
>    e. Go through SWEBOK v3 _while_ going through v4 to compare/contrast

We should talk about whether we want to go to this level of review.  I'm wondering if we could just look at the most recent version of SWEBOK?

> 4. Acceptance Tests vs. Acceptance Testing (see [IEEE2017](https://gitlab.cas.mcmaster.ca/crawfs1/TestGen-Thesis-Papers/-/blob/main/IEEE2017.pdf?ref_type=heads)) - is this a level of distinction we should make? Is this relevant?

The difference seems to be between the approach (acceptance testing) and the individual tests in the approach (acceptance tests).  The relevant definition is between testing and tests.  I'm thinking your table will summarize approaches, like acceptance testing, black box testing, regression testing etc.  Every approach is made up of tests, but we don't need to define those as separate rows in the table.

**Comment by samm82** [comment]
For clarification on acceptance testing vs. tests, I guess I made it seem like there was less of a difference than the document(s) imply. An "acceptance test" is "usually performed by the purchaser **on his premises after installation** with the participation of the vendor" (IEEE, 2017, p. 5, _emphasis added_), while "acceptance testing" is used to "determine whether a system satisfies its acceptance criteria and to enable the customer to determine whether to accept the system" (IEEE, 2017, p. 5), but doesn't seem to imply that the product has been _purchased_, which seems to suggest that the former is done after the product is released while the latter is done before....

**Comment by smiths** [comment]
@samm82 I don't think we want to get too deep into subtle differences like whether the acceptance tests happen before or after purchase.  The "acceptance test" is performed by the purchaser, in the definition, but if they don't accept the test, the vendor won't be done; they will have to fix the problem.  I don't think in practical situations there will be much of a difference between before and after purchase.  Even if it is a difference, it feels like it is getting too far from our original purpose to explore these kind of subtle differences.  (I'm not at all sure that the subtle differences were intentional by the SWEBOK authors.  Sometimes people just phrase things a different way with the goal of variety.)

**Comment by JacquesCarette** [comment]
I agree with all that @smiths has said, and have nothing to add. Is there something for me to 'do' to close this? Or should this be solely assigned to @samm82 ?

**Comment by samm82** [comment]
I've been assigning these meeting issues to the three of us, since we will all be at the meeting (or, if not, be interested in what was discussed). These issues are just for tracking the agendas (to help us be focused during the actual meeting) and summarizing the "results" of the meeting (mainly for me to remember what we talked about/decided on and to have a "paper trail" to potentially point to later).

TL;DR: Nothing to do to close this except have the actual meeting on Monday!

**Comment by samm82** [comment]
1. A helpful "rule of Dr. Smith's thumb" for the depth of research is "as deep as necessary" - further research/investigation into something should have a _reason_ behind it before spending time on it. Going into the summer is reasonable, but I shouldn't go past that (which will be a good motivator)!
2. Even though the presentation might not be "A+" quality, it's a good way to improve presentation skills and get feedback (as well as meeting a requirement 😁). A lit review presentation will be fine.
3. Going through the SWEBOK v4 document next seems like a good idea, since SWEBOK was always on my to-do list and there is a revision "deadline" which will allow me to contribute. (These contributions could be noted, potentially to include in a "results" section?) This can be done in tandem with v3, since I can cite both, and finding discrepancies between the two may be useful to us.
4. Noting this discrepancy between acceptance test/testing _may_ be valuable, especially given the "cf." (meaning "compare") to validation testing, but shouldn't be too much of a time sink.

**Comment by samm82** [comment]
A validation test is a "test to determine whether an implemented system fulfils its specified requirements" (IEEE, 2017, p. 499), which seems related to functional testing and/or specification-based testing. This _may_ be worth investigating further, but I'm honestly not sure. I'm tempted to omit this (and "acceptance test") from my table (especially since it is listed as a "test" and not a method of "testing").
